test/cpp/api/optim_baseline.py

```diff
@@ -26,6 +26,9 @@
     "Adam": lambda p: torch.optim.Adam(p, 1.0),
     "Adam_with_weight_decay": lambda p: torch.optim.Adam(p, 1.0, weight_decay=1e-2),
     "Adam_with_weight_decay_and_amsgrad": lambda p: torch.optim.Adam(p, 1.0, weight_decay=1e-6, amsgrad=True),
+    "AdamW": lambda p: torch.optim.AdamW(p, 1.0),
+    "AdamW_without_weight_decay": lambda p: torch.optim.AdamW(p, 1.0, weight_decay=0),
```

Comment:
Can we have a test case with non-zero decay?

Reply:
The default `weight_decay` is set to `1e-2` in the python implementation, and consequently in mine. Should I write a test with custom weight decay `AdamW_with_custom_weight_decay` (naming it `AdamW_with_weight_decay` will be a bit confusing IMO since the default `AdamW` already has a non-zero weight decay)?

Reply:
Ah, I see what you mean, no need to change here then. The default case already have a non-zero weight_decay being tested. Thanks.